app/src/main/java/com/nervousfish/nervousfish/activities/RhythmCreateActivity.java

```diff
@@ -136,21 +135,24 @@ public void onTapClick(final View v) {
      */
     public void onDoneCreatingRhythmClick(final View v) {
         LOGGER.info("Done tapping button clicked");
+        final Profile profile = this.serviceLocator.getDatabase().getProfile();
+        final KeyPair keyPair = profile.getKeyPairs().get(0);
+
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info("Sending my profile with name: {}, public key: {}", profile.getName(), keyPair.getPublicKey());
+        }
+        final Contact myProfileAsContact = new Contact(profile.getName(), new Ed25519Key("Ed25519 key", "73890ien"));
+        final long key = new RhythmCreateActivity.KMeansClusterHelper().getEncryptionKey(this.taps);
         try {
-            final Profile profile = this.serviceLocator.getDatabase().getProfile();
-            final KeyPair keyPair = profile.getKeyPairs().get(0);
-
-            LOGGER.info("Sending my profile with name: {}, public key: {}", profile.getName(), keyPair.getPublicKey().toString());
-            final Contact myProfileAsContact = new Contact(profile.getName(), new Ed25519Key("Ed25519 key", "73890ien"));
-            final long encryptionKey = new RhythmCreateActivity.KMeansClusterHelper().getEncryptionKey(this.taps);
-            this.bluetoothHandler.send(myProfileAsContact, encryptionKey);
-        } catch (final IOException e) {
-            LOGGER.error("Could not send my contact to other device ", e);
+            this.bluetoothHandler.send(myProfileAsContact, key);
+        } catch (final BadPaddingException | IllegalBlockSizeException e) {
+            LOGGER.error("Could not encrypt the contact");
+            throw new EncryptionException(e);
```

Comment:
Neither `BadPaddingException` nor `IllegalBlockSizeException` seem like encrytion exceptions to me... 🤔